src/lib/viz/layer/Layer.ts

```diff
@@ -576,3 +631,12 @@ interface LayerPosition {
   beforeLayerId?: string;
   afterLayerId?: string;
 }
+
+function buildInitialDataState() {
+  return {
+    isFirstTime: true,
```

Comment:
If we join 'isPanning' / 'isZooming'... we might consider using an enum instead of a set of flags, so only 1 state is possible at the same time...

Reply:
Yep, I think it is possible to do so

Reply:
Done in 2 steps if you want to follow the code